### PR TITLE
NullPointerException in documentColor

### LINKED ISSUE #1473: NullPointerException in `documentColor`
Given the XML document:

```xml
<aaa>
  <color hex=></color>
</aaa>
```

and the configuration:

```jsonc
"xml.colors": [
  {
    "expressions": [
      {
        "xpath": "color/@hex"
      }
    ],
    "pattern": "**.xml"
  }
],
```

You will get an NPE logged in the server trace.

### PR DESCRIPTION
**Pull request by angelozerr**
NullPointerException in documentColor

Fixes #1473

### CODE REVIEW

**Review by JessicaJHee — APPROVED** [review]
LGTM!

File: org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/colors/XMLColorsExtensionsTest.java [hunk 1]
```diff
@@ -71,6 +71,23 @@ public void colorOnAttr() throws BadLocationException {
...
 
+	@Test
+	public void colorOnAttrWithoutValue() throws BadLocationException {
+		// Here color is done for item/@color only
+		XMLColorsSettings settings = createXMLColorsSettings();
+		String xml = "<?xml version=\"1.0\" encoding=\"utf-8\"?>\r\n"
+				+ "<resources>\r\n"
+				+ "	<color name=\"opaque_red\">#f00</color>\r\n"
+				+ "	<color name=\"translucent_red\">#80ff0000</color>\r\n"
+				+ "	<color name=>BAD_COLOR</color>\r\n" // <-- here item/@color has no value
```
Comment (datho7561) [inline org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/colors/XMLColorsExtensionsTest.java:82]: Will this test the bug? The issue I had was with having an attribute that's defined as a colour that has no value (eg. `<item color =>`) but that case is not tested here

Comment (angelozerr) [reply org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/colors/XMLColorsExtensionsTest.java:82]: Sorry I forget your feedback. Indeed I will fix it

Comment (angelozerr) [reply org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/colors/XMLColorsExtensionsTest.java]: @datho7561 it should be fixed now

**Pull request merged**